src/commandline_frame.ts

```diff
@@ -249,6 +262,9 @@ commandline_state.clInput.addEventListener("input", () => {
         if (exstr !== commandline_state.clInput.value) return
 
         onInputPromise = refresh_completions(exstr)
+        onInputPromise.then(() => {
+            contentState.cmdline_filter = exstr
+        })
```

Comment:
Wouldn't it be conceptually simpler to put `onInputPromise` ins `QUEUE`? Then you avoid the need to sync contentState with the page's content script.

Reply:
Perhaps, but we then have to have all commands wait for every input promise. With this PR currently only commands that need to wait for completions (and all ones after it) are delayed.

The `contentState` stuff happens within a single tab so I think it probably has less overhead than you might imagine.

I'll give it a go and see what happens.